### PR TITLE
WIP: Importing from shared dir

### LINKED ISSUE #336: Extract My profile screen components to shared dir

### PR DESCRIPTION
**Pull request by pcowgill**
### Issue link

https://github.com/tasitlabs/tasit/issues/336

### Auto-close the issue?

Closes #336 

### Types of changes

Technical debt (a code change that doesn't fix a bug or add a feature but makes something clearer for devs)

### Notes

This seems like it should be working based on comments I found when googling like this one https://github.com/facebook/react-native/issues/12241#issuecomment-300879473, but it isn't working yet.

### CODE REVIEW

**Comment by pcowgill** [comment]
This might be related https://github.com/expo/expo/commit/1c24b4e17b3c8e02a3d3ec891b0eabca58c8ae18

**Comment by marcelomorgado** [comment]
Do you want that I take a look on this?

**Comment by pcowgill** [comment]
> Do you want that I take a look on this?

@marcelomorgado Sure, that would be great! Thanks!

**Comment by pcowgill** [comment]
@marcelomorgado This is the clearest reference repo I found, where there's an expo app that uses a shared component https://github.com/MPiwowarski/react-native-adjacent-imports

**Comment by pcowgill** [comment]
> @marcelomorgado This is the clearest reference repo I found, where there's an expo app that uses a shared component https://github.com/MPiwowarski/react-native-adjacent-imports

I found it here https://github.com/facebook/react-native/issues/12241#issuecomment-365943162

File: decentraland/metro.config.js [hunk 1]
```diff
@@ -1,6 +1,14 @@
 const path = require("path");
 
 module.exports = {
+  // Both lines extracted from: https://medium.com/react-native-training/sharing-code-between-react-web-and-react-native-applications-7f451af26378
+  // Tech-debt: Use same var for the shared dir (Why using resolve instead of join?)
```
Comment (pcowgill) [inline decentraland/metro.config.js]: Having looked at the `path` docs, either `resolve` or `join` is a perfectly sensible approach to achieve the same thing

Comment (pcowgill) [reply decentraland/metro.config.js]: What do you mean by `Use same var for the shared dir`?

**Comment by marcelomorgado** [comment]
Now, we're getting this error (from expo-cli):
`Unable to resolve "@babel/runtime/helpers/interopRequireDefault" from "../shared/components/presentational/Button.js"`

File: decentraland/metro.config.js (hunk 1 above)
Comment (marcelomorgado) [reply decentraland/metro.config.js]: You've created `var shared = path.join(__dirname, "../shared");`. I've introduced `  watchFolders: [path.resolve(__dirname, "../shared")],`, I've meant to used `shared` variable one `watchFolders` too if is possible.

Comment (pcowgill) [reply decentraland/metro.config.js]: Ah ok, understood!

**Comment by pcowgill** [comment]
> Unable to resolve "@babel/runtime/helpers/interopRequireDefault"

@marcelomorgado Did you do `npm i` in the `shared` dir before running the `decentraland` app when you saw this?

**Comment by pcowgill** [comment]
@marcelomorgado I'm getting this naming collision for `react-native` which is a dep of both the `shared` dir and the `decentraland` dir, but that's too be expected.

```
[16:47:11] (node:64828) UnhandledPromiseRejectionWarning: Error: jest-haste-map: @providesModule naming collision:
[16:47:11]   Duplicate module name: react-native
[16:47:11]   Paths: /Users/paulcowgill/Code/tasitlabs/tasit/shared/node_modules/react-native/package.json collides with /Users/paulcowgill/Code/tasitlabs/tasit/decentraland/node_modules/react-native/package.json
```

I know I saw someone handling that in one of the threads I looked at while researching this. I'll look for that.

Side note: If there are enough shared deps we might want lerna in this repo too https://github.com/tasitlabs/tasit/issues/374

**Comment by pcowgill** [comment]
I'm thinking about splitting the deps up like this. At this point almost all of hte deps can probably go in the `shared` dir unless we're really sure it's `decentraland`-specific or a now-debatable choice we might move away from for tech debt reasons like redux.

Screenshot 2019-05-09 17 04 41

**Comment by pcowgill** [comment]
Or maybe I spoke too soon:

```
[17:06:44] Error: Can't find react-native in package.json dependencies

[17:06:44] Error: React Native is not installed. Please run `npm install` in your project directory.

[17:06:44] Couldn't start project. Please fix the errors and restart the project.

[17:06:44] Set EXPO_DEBUG=true in your env to view the stack trace.
```

I'll keep researching how others have done this

**Comment by pcowgill** [comment]
This was the related comment I remembered https://github.com/react-community/create-react-native-app/issues/232#issuecomment-343717596

**Comment by pcowgill** [comment]
`metro` changed their config format, but this suggestion seems like it will work:

https://github.com/facebook/metro/issues/7#issuecomment-430008657

**Comment by pcowgill** [comment]
> `metro` changed their config format, but this suggestion seems like it will work:
> 
> [facebook/metro#7 (comment)](https://github.com/facebook/metro/issues/7#issuecomment-430008657)

I think this is the right direction to be headed in, but it hasn't solved the issue of complaining about duplicated `react` and `react-native` deps just yet.

**Comment by pcowgill** [comment]
> > Unable to resolve "@babel/runtime/helpers/interopRequireDefault"
> 
> @marcelomorgado Did you do `npm i` in the `shared` dir before running the `decentraland` app when you saw this?

@marcelomorgado I'm not seeing the babel issue - does running `npm i` address it for you?

**Comment by marcelomorgado** [comment]
> > > Unable to resolve "@babel/runtime/helpers/interopRequireDefault"
> > 
> > 
> > @marcelomorgado Did you do `npm i` in the `shared` dir before running the `decentraland` app when you saw this?
> 
> @marcelomorgado I'm not seeing the babel issue - does running `npm i` address it for you?

I'm getting the same error:

```sh
[16:09:04] Error: Can't find react-native in package.json dependencies

[16:09:04] Error: React Native is not installed. Please run `npm install` in your project directory.

[16:09:04] Couldn't start project. Please fix the errors and restart the project.

[16:09:04] Set EXPO_DEBUG=true in your env to view the stack trace.
```

**Comment by pcowgill** [comment]
> > > > Unable to resolve "@babel/runtime/helpers/interopRequireDefault"
> > > 
> > > 
> > > @marcelomorgado Did you do `npm i` in the `shared` dir before running the `decentraland` app when you saw this?
> > 
> > 
> > @marcelomorgado I'm not seeing the babel issue - does running `npm i` address it for you?
> 
> I'm getting the same error:
> 
> ```shell
> [16:09:04] Error: Can't find react-native in package.json dependencies
> 
> [16:09:04] Error: React Native is not installed. Please run `npm install` in your project directory.
> 
> [16:09:04] Couldn't start project. Please fix the errors and restart the project.
> 
> [16:09:04] Set EXPO_DEBUG=true in your env to view the stack trace.
> ```

If I include react-native as a dep in both `decentraland` and `shared`, that issue goes away, but then I get the duplicated provides module error. Which I thought `extraNodeModules` would address

**Pull request closed without merging**